homeassistant/components/enigma2/media_player.py

```diff
@@ -80,49 +71,67 @@ async def async_setup_platform(
-        config[CONF_USERNAME] = DEFAULT_USERNAME
-        config[CONF_PASSWORD] = DEFAULT_PASSWORD
-        config[CONF_SSL] = DEFAULT_SSL
-        config[CONF_USE_CHANNEL_ICON] = DEFAULT_USE_CHANNEL_ICON
-        config[CONF_MAC_ADDRESS] = DEFAULT_MAC_ADDRESS
-        config[CONF_DEEP_STANDBY] = DEFAULT_DEEP_STANDBY
-        config[CONF_SOURCE_BOUQUET] = DEFAULT_SOURCE_BOUQUET
-
-    base_url = URL.build(
-        scheme="https" if config[CONF_SSL] else "http",
-        host=config[CONF_HOST],
-        port=config.get(CONF_PORT),
-        user=config.get(CONF_USERNAME),
-        password=config.get(CONF_PASSWORD),
+
+    host = config[CONF_HOST]
+
+    async_create_issue(
+        hass,
+        HOMEASSISTANT_DOMAIN,
+        f"deprecated_yaml_{DOMAIN}",
+        breaks_in_ha_version="2024.7.0",
+        is_fixable=False,
+        is_persistent=False,
+        issue_domain=DOMAIN,
+        severity=IssueSeverity.WARNING,
+        translation_key="deprecated_yaml",
+        translation_placeholders={
+            "domain": DOMAIN,
+            "integration_title": "Enigma2",
+        },
     )
 
-    session = async_create_clientsession(hass, verify_ssl=False, base_url=base_url)
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.options[CONF_HOST] == host:
+            return
 
-    device = OpenWebIfDevice(
-        host=session,
-        turn_off_to_deep=config.get(CONF_DEEP_STANDBY, False),
-        source_bouquet=config.get(CONF_SOURCE_BOUQUET),
+    entry_data = {
+        CONF_NAME: config.get(CONF_NAME),
+        CONF_HOST: host,
+        CONF_PORT: config.get(CONF_PORT),
+        CONF_USERNAME: config.get(CONF_USERNAME),
+        CONF_PASSWORD: config.get(CONF_PASSWORD),
+        CONF_SSL: config.get(CONF_SSL),
+        CONF_USE_CHANNEL_ICON: config.get(CONF_USE_CHANNEL_ICON),
+        CONF_DEEP_STANDBY: config.get(CONF_DEEP_STANDBY),
+        CONF_SOURCE_BOUQUET: config.get(CONF_SOURCE_BOUQUET),
+    }
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=entry_data
+        )
     )
 
-    try:
-        about = await device.get_about()
-    except ClientConnectorError as err:
-        raise PlatformNotReady from err
 
-    async_add_entities([Enigma2Device(config[CONF_NAME], device, about)])
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Enigma2 media player platform."""
+
+    device: OpenWebIfDevice = hass.data[DOMAIN][entry.entry_id]
+    about = await device.get_about()
+    device.mac_address = about["info"]["ifaces"][0]["mac"]
+    entity = Enigma2Device(entry, device, about)
+    async_add_entities([entity])
 
 
 class Enigma2Device(MediaPlayerEntity):
     """Representation of an Enigma2 box."""
 
     _attr_has_entity_name = True
+    _attr_translation_key = "player"
```

Comment:
This is not correct, it should be like this since the media player entity is the main feature of the device
```suggestion
    _attr_name = None
```

Reply:
Even when I want to add more entities afterwards?

Reply:
Yes, unless the media player entity will no longer the main feature of the Enigma device.
I don't think that's likely though?

Reply:
That's true, the media player is the main feature. Thanks!